content/guides/docker-concepts/running-containers/overriding-container-defaults.md

```diff
@@ -0,0 +1,177 @@
+
+Now you might ask, "How can I override these container defaults?" The answer is `docker run` command.
+
+The `docker run` command offers a powerful way to override these defaults and tailor the container's behavior to your liking. The command offers several flags that let you to customize container behavior on the fly.
+
+Here's a few ways you can achieve this:
+
+### Overriding the network ports
+
+Sometimes you might want to use separate database instances for development and testing purposes. Running these database instances on the same port might conflict. You can use the `-p` option in `docker run` to map container ports to host ports, allowing you to run the multiple instances of the container without any conflict.
+
+```console
+$ docker run -d -p HOST_PORT:CONTAINER_PORT my_image
+```
+
+### Setting environment variables
+
+This option sets an environment variable `foo` inside the container with the value `bar`.
+
+```console
+$ docker run -e foo=bar my_image env
+```
+
+You will see output like the following:
+
+```console
+HOSTNAME=2042f2e6ebe4
+foo=bar
+```
+
+> The `.env` file acts as a convenient way to set environment variables for your Docker containers without cluttering your command line with numerous -e flags. To use a `.env` file, you can pass `--env-file` option with the `docker run` command.
+
+### Restricting the container to consume the resources
+
+You can use the `--memory` and `--cpus` flags with the `docker run` command to restrict how much CPU and memory a container can use. For example, you can set a memory limit for the Python API container, preventing it from consuming excessive resources on your host. Here's the command:
+
+```console
+$ docker run --memory="512m" --cpus="0.5" my_image
+ ```
+
+This command limits container memory usage to 512 megabytes and defines the CPU quota of 0.5 for half a core.
+
+> **Monitor the real-time resource usage**
+>
+> You can use the `docker stats` command to monitor the real-time resource usage of running containers. This helps you understand whether the allocated resources are sufficient or need adjustment.
+{ .tip }
+
+By effectively using these `docker run` flags, you can tailor your containerized application's behaviour to fit your specific requirements!
+
+## Try it now
+
+In this hands-on guide, you'll see how to use the `docker run` command to override the container defaults.
+
+1. [Download and install](https://www.docker.com/products/docker-desktop/) Docker Desktop.
```

Comment:
Either remove the numbered list here, or add this as the first step in the section below.
```suggestion
1. [Download and install](/get-docker/) Docker Desktop.
```